src/Writers/HTMLWriter.jl

```diff
@@ -1608,8 +1613,17 @@ mdconvert(b::Markdown.Bold, parent; kwargs...) = Tag(:strong)(mdconvert(b.text,
 function mdconvert(c::Markdown.Code, parent::MDBlockContext; kwargs...)
     @tags pre code
     language = Utilities.codelang(c.language)
-    language = isempty(language) ? "none" : language
-    pre(code[".language-$(language)"](c.code))
+    class = isempty(language) ? ".nohighlight" : ".language-$(language)"
+    if language == "julia-repl"
+        input = IOBuffer(c.code)
+        printer = AnsiColoredPrinters.HTMLPrinter(input)
+        p = repr(MIME"text/html"(), printer)
+        # TODO: add support for specifying root tag in AnsiColoredPrinters
+        out = replace(replace(p, r"^<pre>\n" => "<code class=\"language-julia-repl\">"), r"/pre>$" => "/code>")
+        return pre(Tag(Symbol("#RAW#"))(out))
```

Comment:
Currently you get a `String` out of AnsiColoredPrinters, right? Could we return a `DOM` tree instead? AnsiColoredPrinters would probably need some API for user-defined printers though?

Reply:
Returning a DOM tree is technically possible, but as you've noticed, there are two problems.
One is that the basic concept of AnsiColoredPrinters is the lazy printing using 3-arg `show`. Therefore, the output can be binary data instead of `String`, but it must be handled by `print`/`write`. That means we need to create some quite new API for the `DOM`.
The other is that `DOM` is a local type of Documenter.jl. If we use the general and portable `DOM` notation instead of the local type, it should be exactly "text/html". :stuck_out_tongue_closed_eyes:

Implementing a user-defined printer isn't hard if you use the private methods of AsiColoredPrinters, but refining the APIs to make them public is a bit of a hassle. :sweat_smile:

If there is a specific benefit to the authors and readers of the documents by expressing it in a DOM tree, I'd like to consider supporting it.

Reply:
It would not be strictly necessary, but from a maintenance perspective, but it's probably a good idea if all the HTML generation happens in one place (in the `DOM` module). Also, once we get to `LaTeXWriter`, I think it would be good if Documenter would have full control over what LaTeX gets generated, in case we need to change that.

I haven't looked at the package very closely, but would it be possible to have the printer be an iterator that iterates over sections of characters that share the same styling or something along those lines? That way generating the correct array of `DOM` `<span>`s should be a matter of a few lines of code in Documenter.

Reply:
The ANSI escape codes are a bit cumbersome to manage the stack in a tree model like HTML, because the start and end codes aren't necessarily one-to-one. Also, an HTML-specific issue is that the colors except the system 16 colors should be assigned in the `style` attribute directly.

So I'd like to add a mechanism to call a callback method "just before" writing an HTML tag. If we call `String(take!(io))` in the callback method, we should be able to get the text from the previous tag to the current position.

Reply:
The `HTMLPrinter` has a new `callback` option.
https://kimikage.github.io/ANSIColoredPrinters.jl/dev/output-formats/#html

So, the "text/plain" output is now DOMified (in a somewhat tricky way).